File: src/main/java/com/chaottic/nbt/number/LongTag.java
package com.chaottic.nbt.number;

import java.io.DataInput;
import java.io.IOException;

public record LongTag(long l) implements NumberTag {
    @Override
    public byte getAsByte() {
        return (byte) l;
    }

    @Override
    public short getAsShort() {
        return (short) l;
    }

    @Override
    public int getAsInt() {
        return (int) l;
    }

    @Override
    public long getAsLong() {
        return l;
    }

    @Override
    public float getAsFloat() {
        return l;
    }

    @Override
    public double getAsDouble() {
        return l;
    }

    public static LongTag createLongTag(DataInput dataInput) throws IOException {
        return new LongTag(dataInput.readLong());
    }
}

File: src/main/kotlin/com/chaottic/nbt/number/LongTag.kt
package com.chaottic.nbt.number

import com.chaottic.nbt.TagType
import java.io.DataInput

data class LongTag(private val l: Long) : NumberTag {

	override fun asByte() = l.toByte()
	override fun asShort() = l.toShort()
	override fun asInt() = l.toInt()
	override fun asLong() = l
	override fun asFloat() = l.toFloat()
	override fun asDouble() = l.toDouble()

	companion object : TagType<LongTag> {

		override fun createTag(dataInput: DataInput) = LongTag(dataInput.readLong())
	}
}
